tests/rest/media/v1/test_filepath.py

```diff
@@ -486,3 +487,41 @@ def _test_path_validation(
                         f"{value!r} unexpectedly passed validation: "
                         f"{method} returned {path_or_list!r}"
                     )
+
+    def test_symlink(self):
```

Comment:
Should we add tests for some of the edge cases using `..` and `.`?

Reply:
There's another set of checks (not touched in this PR) that inspects individual path components and rejects `..` and `.`, which makes it hard to test the unhappy paths in the decorator.

At the time I wasn't sure how to test the decorator in isolation but I think I have an idea.

Reply:
Ah, I must have missed that! I'll double check.

Reply:
I've added a couple of negative test cases